src/main/java/com/rackspace/salus/event/ingest/services/IngestService.java

```diff
@@ -112,12 +115,16 @@ public void consumeMetric(UniversalMetricFrame metric) {
     pointBuilder.tag(Tags.MONITORING_SYSTEM, metric.getMonitoringSystem().toString());
 
     metric.getMetricsList().stream().forEach(val -> {
-      if(!StringUtils.isEmpty(val.getString())) {
-        pointBuilder.addField(val.getName(), val.getString());
-      } else if(val.getFloat()!=0.0) {
-        pointBuilder.addField(val.getName(), val.getFloat());
-      } else if(val.getInt()!=0) {
-        pointBuilder.addField(val.getName(), val.getInt());
+      Metric.ValueCase valueCase = Metric.ValueCase.forNumber(val.getValueCase().getNumber());
+      switch (valueCase) {
+        case INT:
+          pointBuilder.addField(val.getName(), val.getInt());
+          break;
+        case FLOAT:
+          pointBuilder.addField(val.getName(), val.getFloat());
+          break;
+        case STRING:
+          pointBuilder.addField(val.getName(), val.getString());
```

Comment:
To be safe, like if a new case is added later:
```suggestion
        case STRING:
          pointBuilder.addField(val.getName(), val.getString());
          break;
```